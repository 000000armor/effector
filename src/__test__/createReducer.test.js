// @flow

import {
  createReducer,
  createDomain, 
  type Domain, 
  type Event, 
  type Reducer,
  rootDomain, 
  effectorMiddleware
} from '..'

type State = {foo: 'bar'}

test('reducer', () => {
  const domain = rootDomain()
  const event1: Event<'ev', State> = domain.event('event1')
  const event2: Event<'ev2', State> = domain.event('event2')
  const reset: Event<void, State> = domain.event('reset')

  const reducer: Reducer<{
    actions: string[]
  }> = createReducer({actions: ['bar']})
    .on([event1, event2], (state, payload) => ({
      actions: [...state.actions, payload]
    }))
    .reset(reset)
  
  const state1 = reducer(undefined, event1('ev').raw())
  expect(state1).toMatchSnapshot()
  const state2 = reducer(state1, event2('ev2').raw())
  expect(state2).toMatchSnapshot()
  const state3 = reducer(state1, reset().raw())
  expect(state3).toMatchSnapshot()
})