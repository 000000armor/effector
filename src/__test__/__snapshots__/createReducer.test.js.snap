// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer 1`] = `
Object {
  "actions": Array [
    "bar",
    "ev",
  ],
}
`;

exports[`reducer 2`] = `
Object {
  "actions": Array [
    "bar",
    "ev",
    "ev2",
  ],
}
`;

exports[`reducer 3`] = `
Object {
  "actions": Array [
    "bar",
  ],
}
`;
